### ISSUE TITLE
Python3 Rewrite

### ISSUE THREAD

**Issue by roshbhatia** [roshbhatia/toggle#1]
Upgrade to Python3

Python2 is approaching its end-of-life. As such, the existing code will become increasingly difficult to maintain and enhance.

Although there are some differences, most of our project consists of calls to libraries which already have a python3 version available. As a result, it should be possible to write a new main loop in python3 and then, without making any significant changes to the logic, adjust the syntax for the remainder of the code so that it will link with the new versions of those libraries

The resulting implementation would be a drop-in replacement for the existing program It would provide the same interfaces and responses, but run within the Python3 infrastructure.

**Comment by roshbhatia** [comment]
Ben and I were able to convert all the src files to python3, currently working on the tests (although it seems like the issues with some of the test had existed before the conversion)

**Comment by roshbhatia** [comment]
Turns out we branched off of master instead of 1.3.x, so deleted our branch and created a new one. Converted all files to python3, but there are some issues that still need fixing.

**Comment by goeland86** [comment]
@roshbhatia can you specify which branch is the correct one? I see `dev-py3` and `dev-python3` in the repository.

**Comment by roshbhatia** [comment]
correct branch is `dev-py3`

**Comment by roshbhatia** [comment]
It seems like @benhg got the rest of the tests to work, waiting on feedback to continue to another story.

**Comment by roshbhatia** [comment]
@Wackerbarth asked us to squash the changes into a single commit before merging into intelligent-agent/toggle, which we did. Having some issues with yapf in the travis test -- need some help on that front.

**Comment by roshbhatia** [comment]
Merged!
